python/lsst/ip/isr/isrTask.py

```diff
@@ -469,220 +642,502 @@ def readIsrData(self, dataRef, rawExposure):
-                self.suspectDetection(ccdExposure, amp)
-                overscanResults = self.overscanCorrection(ccdExposure, amp)
-                overscans.append(overscanResults.overscanImage if overscanResults is not None else None)
+                # Check for fully masked bad amplifiers, and generate masks for SUSPECT and SATURATED values.
+                badAmp = self.maskAmplifier(ccdExposure, amp, defects)
+
+                if self.config.doOverscan and not badAmp:
+                    # Overscan correction on amp-by-amp basis.
+                    overscanResults = self.overscanCorrection(ccdExposure, amp)
+                    self.log.info("Corrected overscan for amplifier %s" % (amp.getName()))
+                    if self.config.qa is not None and self.config.qa.saveStats is True:
+                        if isinstance(overscanResults.overscanFit, float):
+                            qaMedian = overscanResults.overscanFit
+                            qaStdev = float("NaN")
+                        else:
+                            qaStats = afwMath.makeStatistics(overscanResults.overscanFit,
+                                                             afwMath.MEDIAN | afwMath.STDEVCLIP)
+                            qaMedian = qaStats.getValue(afwMath.MEDIAN)
+                            qaStdev = qaStats.getValue(afwMath.STDEVCLIP)
+
+                        self.metadata.set("ISR OSCAN {} MEDIAN".format(amp.getName()), qaMedian)
+                        self.metadata.set("ISR OSCAN {} STDEV".format(amp.getName()), qaStdev)
+                        self.log.info("  Overscan stats for amplifer %s: %f +/- %f" %
+                                      (amp.getName(), qaMedian, qaStdev))
+                        ccdExposure.getMetadata().set('OVERSCAN', "Overscan corrected")
+                else:
+                    self.log.warn("Amplifier %s is bad." % (amp.getName()))
+                    overscanResults = None
+
+                overscans.append(overscanResults if overscanResults is not None else None)
             else:
-                overscans.append(None)
+                self.log.info("Skipped OSCAN")
 
-        if self.config.doCrosstalk:
-            self.crosstalk.run(ccdExposure, crosstalkSources)
+        if self.config.doCrosstalk and self.config.doCrosstalkBeforeAssemble:
+            self.log.info("Applying crosstalk correction.")
+            self.crosstalk.run(ccdExposure, crosstalkSources=crosstalkSources)
+            self.debugView(ccdExposure, "doCrosstalk")
 
         if self.config.doAssembleCcd:
+            self.log.info("Assembling CCD from amplifiers")
             ccdExposure = self.assembleCcd.assembleCcd(ccdExposure)
+
             if self.config.expectWcs and not ccdExposure.getWcs():
                 self.log.warn("No WCS found in input exposure")
+            self.debugView(ccdExposure, "doAssembleCcd")
+
+        # CZW: push these into output struct, add to docstring
```

Comment:
I don't understand this note.  If there is no config to do the Oss writing, shouldn't this just go away?